src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20240402135931_EES5037_AddDataSetFileColumnsToFilesTable.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Migrations.ContentMigrations
+{
+    /// <inheritdoc />
+    public partial class EES5037_AddDataSetFileColumnsToFilesTable : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<Guid>(
+                name: "DataSetFileId",
+                table: "Files",
+                type: "uniqueidentifier",
+                nullable: true);
+
+            migrationBuilder.AddColumn<int>(
+                name: "DataSetFileVersion",
+                table: "Files",
+                type: "int",
+                nullable: true);
+
+            migrationBuilder.Sql("""
+                                 UPDATE Files
+                                 SET DataSetFileId = NEWID(), DataSetFileVersion = 0
+                                 WHERE Type = 'Data'
+                                 """);
```

Comment:
I'm just wondering if we should think a bit more closely about this because I'm not convinced it's compatible with a future vision to migrate Files and ReleaseFiles of type data to `DataSetFile` and `DataSetFileVersion` entities if we ever want to do that.

Although we wouldn't easily be able to link together files which replaced each other as versions of the same data set, files that exist on multiple release versions have versionable attributes on `ReleaseFile` like `Name` (Summary),  `Summary` (Data guidance), and `Order` (order on the release version).

So I guess it's a question of whether in future we'd expect `DataSetFileVersion` to have the Name, Summary and Order  along with an association to the `ReleaseVersion`? I've not really thought about it too much but I think it could work like this so that data catalogue's `ListDataSets` in future searches `DataSetFileVersion` entities rather than `ReleaseFiles` and all the attributes needed for that view are either on the `DataSetFileVersion` or the parent `DataSetFile` just like they are currently on `ReleaseFile` and `File`. We'd probably create the versions in order of the release versions order that they exist for.

I'm just wary that this change is saying all existing files are all a first version and that the version is only incremented when the file is replaced, but as soon as we have that sequence of versions aren't we causing more confusion for the future if we want the ReleaseFile versionable info that we already have to migrated to become versions?

Reply:
We discussed this in a meeting. See EES-5050 for details.